app/controllers/api/base_controller/renderer.rb

```diff
@@ -166,7 +166,9 @@ def virtual_attribute_search(resource, attribute)
         if resource.class < ApplicationRecord
           # is relation in 'attribute' variable plural in the model class (from 'resource.class') ?
           if [:has_many, :has_and_belongs_to_many].include?(resource.class.reflection_with_virtual(attribute).try(:macro))
-            Rbac.filtered(resource.public_send(attribute))
+            resource_attr = resource.public_send(attribute)
+            return resource_attr if resource_attr.kind_of?(Array) && resource_attr.first.kind_of?(String)
```

Comment:
it is because Service#power_states is `:virtual_has_many` relation what is a custom method and I think that it can return any data type In general, so I think the condition have to be built on fact `what we can pass to RBAC`,
we can pass to Rbac.filtered(Vm):
1. model (`Rbac.filtered(Vm)`)
2. relation, scope (Vm.with_ownership.class
Vm::ActiveRecord_Relation < ActiveRecord::Relation)
3. the array of ids
4. the array of ActiveRecord objects

and I think for this case is relevant point 4.

@jntullo what do you think about it?